controllers/main.py

```diff
@@ -0,0 +1,40 @@
+from odoo import http
+from odoo.http import request
+class PurchaseOrderController(http.Controller):
+    
+    @http.route('/archive/', auth='public', type='json')
+    def purchase_order_archive(self, **kw):
+        if kw.get('method') != 'archive':
+            return {
+                    "archived_orders": False,
+                    "code": 400,
+                    "message": f"Method is not supported: '{kw.get('method')}'"}
+
+        po_obj = request.env['purchase.order']
+        po_ids = kw.get('orders')
+
+        if po_ids:
+            check = list(set(po_ids)-set(po_obj.sudo().search([]).ids))
```

Comment:
Please do not use `search` function with empty domain. You can use:
```request.env["purchase.order"].sudo().browse(params.get("orders", []))```
and then try to use `exist` function to perform your check